android/app/src/main/java/com/chorestory/app/CreateClanActivity.java

```diff
@@ -44,8 +63,35 @@ public void onClick(View v) {
                 username = usernameEditText.getText().toString();
                 password = passwordEditText.getText().toString();
 
-                // TODO: verification
-                //  if successful register new clan & user, else display snackbar/toast
+                RegisterRequest registerRequest = new RegisterRequest(clanName, username, password);
+                Call<LoginResponse> registerQuery = retrofitInterface.register(registerRequest);
+
+                registerQuery.enqueue(new Callback<LoginResponse>() {
+                    @Override
+                    public void onResponse(Call<LoginResponse> call, Response<LoginResponse> response) {
+                        if (response.isSuccessful()) {
+                            String token = response.body().getToken();
+
+                            // TODO: Store the token properly here
+                            tokenHandler.setToken(token);
+
+                            // TODO: Navigate to profile page
+                        } else{
+                            // TODO: use Snackbar instead; move existing view up when Snackbar appears
+                            toast = Toast.makeText(CreateClanActivity.this,
+                                    "Something went wrong!",
+                                    Toast.LENGTH_LONG);
+                            toast.setGravity(Gravity.CENTER_HORIZONTAL, 0, 400);
+                            toast.show();
+                            signUpButton.setEnabled(true);
+                        }
+                    }
+                    @Override
+                    public void onFailure(Call<LoginResponse> call, Throwable t) {
+                        toast.show();
```

Comment:
need to make text and set gravity first
see line 81

Reply:
would negative gravity make it fly upwards forever, if so can we do that?